### PR TITLE
[DOCS] Reformat shrink index API docs

### PR DESCRIPTION
**Pull request by jrodewig**
Reformats the shrink API docs to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to #43765.

### CODE REVIEW

File: docs/reference/indices/shrink-index.asciidoc [hunk 1]
```diff
@@ -4,36 +4,30 @@
...
+`PUT /<index>/_shrink/<target-index>`
+
+
+[[shrink-index-api-prereqs]]
+==== {api-prereq-title}
 
 In order to shrink an index, the index must be marked as read-only, and a
 (primary or replica) copy of every shard in the index must be relocated to the
-same node and have <<cluster-health,health>> `green`.
+same node and have a <<cluster-health, cluster health>> status of `green`.
```
Comment (debadair) [inline docs/reference/indices/shrink-index.asciidoc]: This reads awkardly--the subject is "a copy of every shard", but the last clause is referring to the node status. I'd probably make this a list:

Before you can shrink an index:

* The index must be read-only.
* A copy of every shard in the index must reside on the same node.
* The <<cluster-health, cluster health>> status must be `green`.

File: docs/reference/indices/shrink-index.asciidoc [hunk 2]
```diff
@@ -4,36 +4,30 @@
...
 
 These two conditions can be achieved with the following request:
```
Comment (debadair) [inline docs/reference/indices/shrink-index.asciidoc]: There are really three conditions.

File: docs/reference/indices/shrink-index.asciidoc [hunk 3]
```diff
@@ -60,15 +54,44 @@ with the <<cat-recovery,`_cat recovery` API>>, or the <<cluster-health,
...
+
+* Then it hard-links segments from the source index into the target index. (If
+  the file system doesn't support hard-linking, then all segments are copied
+  into the new index, which is a much more time consuming process. Also if using 
+  multiple data paths, shards on different data paths require a full copy of 
+  segment files if they are not on the same disk since hardlinks don’t work across
+  disks)
+
+* Finally, it recovers the target index as though it were a closed index which
+  had just been re-opened.
```
Comment (debadair) [inline docs/reference/indices/shrink-index.asciidoc]: Referring to "Shrinking" as "it" strikes me as odd. Maybe:

A shrink operation:

1. Creates the target index...
2. Hard links...
3. Recovers the target index...

File: docs/reference/indices/shrink-index.asciidoc [hunk 4]
```diff
@@ -153,9 +177,35 @@ shrink process begins. When the shrink operation completes, the shard will
 become `active`. At that  point, Elasticsearch will try to allocate any
 replicas and may decide to relocate the primary shard to another node.
 
-[float]
-==== Wait For Active Shards
+
+[[shrink-wait-active-shards]]
+===== Wait For active shards
```
Comment (debadair) [inline docs/reference/indices/shrink-index.asciidoc]: Lowercase for?

**Review by debadair — APPROVED** [review]
A couple minor editorial comments, but other than that LGTM.

File: docs/reference/indices/shrink-index.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/indices/shrink-index.asciidoc]: Fixed with e6f9b25b736.

File: docs/reference/indices/shrink-index.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/indices/shrink-index.asciidoc]: Fixed with e6f9b25b736.

File: docs/reference/indices/shrink-index.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/indices/shrink-index.asciidoc]: Fixed with e6f9b25b736.

File: docs/reference/indices/shrink-index.asciidoc (hunk 4 above)
Comment (jrodewig) [reply docs/reference/indices/shrink-index.asciidoc]: Fixed with e6f9b25b736.

**Comment by jrodewig** [comment]
Thanks @debadair!

**Pull request merged**